arm-gateway/2016-06-01/swagger/gateway.json

```diff
@@ -0,0 +1,442 @@
+          "description": "Gets the resource type."
+        },
+        "location": {
+          "type": "string",
+          "description": "The resource location."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The resource tags."
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "The base resource type."
+    },
+    "SubResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The resource id."
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "The sub resource type."
+    },
+    "Object": {
+      "type": "object",
+      "properties": {}
+    },
+    "ResourceReference": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The resource id."
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the resource name."
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the resource type."
+        }
+      },
+      "description": "The resource reference."
+    },
+    "Gateway": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/GatewayProperties",
+          "description": "The gateway properties."
+        }
+      },
+      "description": "The gateway type.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ]
+    },
+    "GatewayProperties": {
+      "type": "object",
+      "properties": {
+        "connectionGatewayInstallation": {
+          "$ref": "#/definitions/ConnectionGatewayInstallationProperties",
+          "readOnly": true,
+          "description": "Gets the gateway installation properties."
+        },
+        "contactInformation": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the contact information."
+        },
+        "displayName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the display name of the installation."
+        },
+        "machineName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The state."
+        },
+        "status": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the status."
+        },
+        "backendUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the backend URI of installation."
+        }
+      },
+      "description": "The gateway properties."
+    },
+    "ConnectionGatewayInstallationProperties": {
+      "type": "object",
```

Comment:
this model looks a lot like a resource? is ConnectionGatewayInstallation an ARM resource?